File: LatihanInheritanceC++/Kendaraan/Kendaraan.cpp
#include "Kendaraan.hpp"

Kendaraan::Kendaraan()
{
    this->nomorKendaraan = 0;
    this->merk = "XXX";
    this->tahunKeluaran = 0;
}

Kendaraan::Kendaraan(int nomorKendaraan, string merk, int tahunKeluaran)
{
    this->nomorKendaraan = nomorKendaraan;
    this->merk = merk;
    this->tahunKeluaran = tahunKeluaran;
}

Kendaraan::Kendaraan(const Kendaraan &k)
{
    this->nomorKendaraan = k.nomorKendaraan;
    this->merk = k.merk;
    this->tahunKeluaran = k.tahunKeluaran;
}

Kendaraan::~Kendaraan()
{ /*tidak butuh*/
}

int Kendaraan :: BiayaSewa(int lamaSewa)
{return 0;}

void Kendaraan::PrintInfo()
{
    cout << "Nomor Kendaraan: " << this->nomorKendaraan << endl;
    cout << "Merk: " << this->merk << endl;
    cout << "Tahun Keluaran: " << this->tahunKeluaran << endl;
}

File: LatihanInheritanceJava/src/Kendaraan.java
public abstract class Kendaraan {
    protected int nomorKendaraan;
    protected int tahunKeluaran;
    protected String merk;

    public Kendaraan() {
        this.nomorKendaraan = 0;
        this.merk = "XXX";
        this.tahunKeluaran = 0;
    }

    public Kendaraan(int nomorKendaraan, String merk, int tahunKeluaran) {
        this.nomorKendaraan = nomorKendaraan;
        this.merk = merk;
        this.tahunKeluaran = tahunKeluaran;
    }

    public Kendaraan(Kendaraan k) {
        this.nomorKendaraan = k.nomorKendaraan;
        this.merk = k.merk;
        this.tahunKeluaran = k.tahunKeluaran;
    }

    public abstract long BiayaSewa(int lamaSewa);

    public void printInfo() {
        System.out.printf("Nomor Kendaraan: %d\n", this.nomorKendaraan);
        System.out.printf("Merk: %s\n", this.merk);
        System.out.printf("Tahun Keluaran: %d\n", this.tahunKeluaran);
    }
}
